InstallScripts/]/google-cloud-sdk/help/man/man1/gcloud.1
.TH "GCLOUD " "1" "" "" ""
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.nh
.ad l
.SH "NAME"
.HP
gcloud \- manage Google Cloud Platform resources and developer workflow
.SH "SYNOPSIS"
.HP
gcloud\ \fIGROUP\fR\ | \ \fICOMMAND\fR [\fB\-\-format\fR\ \fIFORMAT\fR] [\fB\-\-help\fR] [\fB\-\-project\fR\ \fIPROJECT_ID\fR] [\fB\-\-quiet\fR,\ \fB\-q\fR] [\fB\-\-trace\-token\fR\ \fITRACE_TOKEN\fR] [\fB\-\-user\-output\-enabled\fR\ \fIUSER_OUTPUT_ENABLED\fR] [\fB\-\-verbosity\fR\ \fIVERBOSITY\fR] [\fB\-h\fR] [\fB\-v\fR,\ \fB\-\-version\fR]
.SH "DESCRIPTION"
.sp
The \fBgcloud\fR CLI manages authentication, local configuration, developer workflow, and interactions with the Google Cloud Platform APIs\&.
.SH "FLAGS"
.PP
\fB\-\-format\fR \fIFORMAT\fR
.RS 4
Specify a format for printed output\&. By default, a command\-specific human\-friendly output format is used\&. Setting this flag to one of the available options will serialize the result of the command in the chosen format and print it to stdout\&. Supported formats are:
json,
text,
yaml\&.
.RE
.PP
\fB\-\-help\fR
.RS 4
Display detailed help\&.
.RE
.PP
\fB\-\-project\fR \fIPROJECT_ID\fR
.RS 4
The Google Cloud Platform project name to use for this invocation\&. If omitted then the current project is assumed\&.
.RE
.PP
\fB\-\-quiet\fR, \fB\-q\fR
.RS 4
Disable all interactive prompts when running gcloud commands\&. If input is required, defaults will be used, or an error will be raised\&.
.RE
.PP
\fB\-\-trace\-token\fR \fITRACE_TOKEN\fR
.RS 4
Token used to route traces of service requests for investigation of issues\&.
.RE
.PP
\fB\-\-user\-output\-enabled\fR \fIUSER_OUTPUT_ENABLED\fR
.RS 4
Control whether user intended output is printed to the console\&. (true/false)
.RE
.PP
\fB\-\-verbosity\fR \fIVERBOSITY\fR
.RS 4
Override the default verbosity for this command\&. This must be a standard logging verbosity level: [debug, info, warning, error, critical, none] (Default: [warning])\&.
.RE
.PP
\fB\-h\fR
.RS 4
Print a summary help and exit\&.
.RE
.PP
\fB\-v\fR, \fB\-\-version\fR
.RS 4
Print version information\&.
.RE
.SH "GROUPS"
.sp
\fIGROUP\fR is one of the following:
.PP
\fBauth\fR
.RS 4
Manage oauth2 credentials for the Google Cloud SDK\&.
.RE
.PP
\fBcomponents\fR
.RS 4
List, install, update, or remove Google Cloud SDK components or packages\&.
.RE
.PP
\fBcompute\fR
.RS 4
Read and manipulate Google Compute Engine resources\&.
.RE
.PP
\fBconfig\fR
.RS 4
View and edit Google Cloud SDK properties\&.
.RE
.PP
\fBdns\fR
.RS 4
Manage Cloud DNS\&.
.RE
.PP
\fBpreview\fR
.RS 4
Manage Preview CLI command groups\&.
.RE
.PP
\fBsql\fR
.RS 4
Manage Cloud SQL databases\&.
.RE
.SH "COMMANDS"
.sp
\fICOMMAND\fR is one of the following:
.PP
\fBhelp\fR
.RS 4
Prints detailed help messages for the specified commands\&.
.RE
.PP
\fBinfo\fR
.RS 4
Display information about the current gcloud environment\&.
.RE
.PP
\fBinit\fR
.RS 4
Create and initialize a gcloud workspace in the current directory\&.
.RE
.PP
\fBversion\fR
.RS 4
Print version information for Cloud SDK components\&.
.RE
.SH "NOTES"
.sp
This command is in the Google Cloud SDK \fBcore\fR component\&. See installing components if it is not installed\&.


InstallScripts/]/google-cloud-sdk/lib/googlecloudsdk/gcloud/gcloud.py
# Copyright 2013 Google Inc. All Rights Reserved.

"""gcloud command line tool."""

import os
import signal
import sys


# Disable stack traces when people kill a command.
def CTRLCHandler(unused_signal, unused_frame):
  """Custom SIGNINT handler.

  Signal handler that doesn't print the stack trace when a command is
  killed by keyboard interupt.
  """
  try:
    log.err.Print('\n\nCommand killed by keyboard interrupt\n')
  except NameError:
    sys.stderr.write('\n\nCommand killed by keyboard interrupt\n')
  # Kill ourselves with SIGINT so our parent can detect that we exited because
  # of a signal. SIG_DFL disables further KeyboardInterrupt exceptions.
  signal.signal(signal.SIGINT, signal.SIG_DFL)
  os.kill(os.getpid(), signal.SIGINT)
  # Just in case the kill failed ...
  sys.exit(1)
signal.signal(signal.SIGINT, CTRLCHandler)


# Enable normal UNIX handling of SIGPIPE to play nice with grep -q, head, etc.
# See https://mail.python.org/pipermail/python-list/2004-June/273297.html and
# http://utcc.utoronto.ca/~cks/space/blog/python/SignalExceptionSurprise
# for more details.
if hasattr(signal, 'SIGPIPE'):
  signal.signal(signal.SIGPIPE, signal.SIG_DFL)


# If we're in a virtualenv, import site packages.
if os.environ.get('VIRTUAL_ENV'):
  # pylint:disable=unused-import
  # pylint:disable=g-import-not-at-top
  import site


def _SetPriorityCloudSDKPath():
  """Put google-cloud-sdk/lib at the beginning of sys.path.

  Modifying sys.path in this way allows us to always use our bundled versions
  of libraries, even when other versions have been installed. It also allows the
  user to install extra libraries that we cannot bundle (ie PyOpenSSL), and
  gcloud commands can use those libraries.
  """

  def _GetRootContainingGoogle():
    path = __file__
    while True:
      parent, here = os.path.split(path)
      if not here:
        break
      if here == 'googlecloudsdk':
        return parent
      path = parent

  module_root = _GetRootContainingGoogle()

  # check if we're already set
  if sys.path and module_root == sys.path[0]:
    return
  sys.path.insert(0, module_root)


def _DoStartupChecks():
  # pylint:disable=g-import-not-at-top
  from googlecloudsdk.core.util import platforms
  if not platforms.PythonVersion().IsSupported():
    sys.exit(1)
  if not platforms.Platform.Current().IsSupported():
    sys.exit(1)

_SetPriorityCloudSDKPath()
_DoStartupChecks()


# pylint:disable=g-import-not-at-top, We want the _SetPriorityCloudSDKPath()
# function to be called before we try to import any CloudSDK modules.
from googlecloudsdk.calliope import cli
from googlecloudsdk.core import config
from googlecloudsdk.core import log
from googlecloudsdk.core import metrics
from googlecloudsdk.core import properties
from googlecloudsdk.core.updater import local_state
from googlecloudsdk.core.updater import update_manager
from googlecloudsdk.core.util import resource_registration

resource_registration.RegisterReleasedAPIs()




def UpdateCheck():
  try:
    update_manager.UpdateManager(warn=False).PerformUpdateCheck()
  # pylint:disable=broad-except, We never want this to escape, ever. Only
  # messages printed should reach the user.
  except Exception:
    pass


def VersionFunc():
  _cli.Execute(['version'])


def CreateCLI():
  """Generates the gcloud CLI."""
  sdk_root = config.Paths().sdk_root
  if sdk_root:
    help_dir = os.path.join(sdk_root, 'help')
  else:
    help_dir = None
  loader = cli.CLILoader(
      name='gcloud',
      command_root_directory=os.path.join(
          cli.GoogleCloudSDKPackageRoot(),
          'gcloud',
          'sdktools',
          'root'),
      allow_non_existing_modules=True,
      version_func=VersionFunc,
      help_func=cli.GetHelp(help_dir))
  pkg_root = cli.GoogleCloudSDKPackageRoot()
  loader.AddModule('auth', os.path.join(pkg_root, 'gcloud', 'sdktools', 'auth'))
  loader.AddModule('bigquery', os.path.join(pkg_root, 'bigquery', 'commands'))
  loader.AddModule('components',
                   os.path.join(pkg_root, 'gcloud', 'sdktools', 'components'))
  loader.AddModule('compute', os.path.join(pkg_root, 'compute', 'subcommands'))
  loader.AddModule('config',
                   os.path.join(pkg_root, 'gcloud', 'sdktools', 'config'))
  loader.AddModule('dns', os.path.join(pkg_root, 'dns', 'dnstools'))
  loader.AddModule('endpoints', os.path.join(pkg_root, 'endpoints', 'commands'))
  loader.AddModule('internal',
                   os.path.join(pkg_root, 'gcloud', 'sdktools', 'internal'))
  loader.AddModule('preview', os.path.join(pkg_root, 'preview', 'commands'))
  # Put app and datastore under preview for now.
  loader.AddModule('preview.analysis',
                   os.path.join(pkg_root, 'analysis', 'commands'))
  loader.AddModule('preview.app',
                   os.path.join(pkg_root, 'appengine', 'app_commands'))
  loader.AddModule('preview.container',
                   os.path.join(pkg_root, 'container', 'commands'))
  loader.AddModule('preview.datastore',
                   os.path.join(pkg_root, 'appengine', 'datastore_commands'))
  loader.AddModule('preview.dns', os.path.join(pkg_root, 'dns', 'commands'))
  loader.AddModule('preview.genomics',
                   os.path.join(pkg_root, 'genomics', 'commands'))
  # TODO(user): Put logging in preview for now.
  loader.AddModule('preview.logging',
                   os.path.join(pkg_root, 'logging', 'commands'))
  loader.AddModule('preview.projects',
                   os.path.join(pkg_root, 'projects', 'commands'))
  loader.AddModule('preview.rolling_updates',
                   os.path.join(
                       pkg_root, 'updater', 'commands', 'rolling_updates'))
  loader.AddModule('preview.test', os.path.join(pkg_root, 'test', 'commands'))
  loader.AddModule('sql', os.path.join(pkg_root, 'sql', 'tools'))

  # Check for updates on shutdown but not for any of the updater commands.
  loader.RegisterPostRunHook(UpdateCheck,
                             exclude_commands=r'gcloud\.components\..*')
  return loader.Generate()

_cli = CreateCLI()


def main():
  # TODO(user): Put a real version number here
  metrics.Executions(
      'gcloud',
      local_state.InstallationState.VersionForInstalledComponent('core'))
  _cli.Execute()

if __name__ == '__main__':
  try:
    main()
  except KeyboardInterrupt:
    CTRLCHandler(None, None)
